third_party/window/window.go

```diff
@@ -0,0 +1,87 @@
+// source: https://github.com/jbrukh/window
+package store
+
+// An array-based moving window; a moving window
+// is a queue with a maximum size and the property
+// that when the size is reached, pushing a new
+// element into the queue causes the head to be
+// popped.
+//
+// You can optimize the amount of slice copying
+// that the MovingWindow will be doing by trading
+// off with space complexity. Namely, the underlying
+// array is allocated with a size that is the
+// multiple of the intended capacity of the queue
+// so that copying is less frequent.
+type MovingWindow struct {
+	arr  []interface{}
```

Comment:
If we're only going to be using this for the StatStore, you might want to use the specific type that we need. The only reason why would be so that we can use `[]hourEntry`, which is contiguous in memory, instead of `[]interface{}` which uses pointers under the covers.

In general, unless each element is very large and thus expensive to swap/move/copy, it's better to store its value directly.